lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -0,0 +1,36 @@
+require "pry"
+
+module RuboCop
+  module Cop
+    module RootCops
+      class FactoryFileName < Cop
+        FILE_NAME_ERROR = "Factory name should match file name (one factory per file)".freeze
+        FACTORY = %i[factory].freeze
+
+        def investigate(processed_source)
+          file_path = processed_source.buffer.name
+          @base_name = File.basename(file_path, ".rb")
+        end
+
+        def on_send(node)
+          method_name = node.method_name
+          return unless _factory_method?(method_name)
+
+          # these 2 lines turn [:factory_name] into "factory_name"
```

Comment:
In general I try to avoid comments that say what the code does.  Usually its a sign that you could name variables or functions better or that the comment is redundant.  It also ends up being something you have to change in lock step with the code, but there's pretty much no good way to enforce that you do that so its hard to ever trust an arbitrary comment you come across.

Reply:
Yeah I was hesitant to add the comment but this parsing syntax has been hard for me to grasp so I guess it was more for myself to reference. But what you said makes a lot of sense, I'll go ahead and remove it.

Reply:
Thank you!